File: client/src/types/player.ts
import {CardUpOrDown, Card} from './card'
import {Coord, Config} from './coord'

export interface Player {
	name: string
	robot: Robot
	hand: CardUpOrDown[]
	board: { [i: number]: CardUpOrDown }
	flagNum: number
	spawn: Coord
}

export function newPlayer(name:string) : Player {
	return {
		name: name,
		hand: [],
		board: {},
		robot: newRobot(),
		flagNum: 0,
		spawn: null,
	}
}

function newRobot() : Robot {
	return {
		damage: 0,
		lives: 0,
		config: null,
	}
}

export interface Robot {
	damage: number
	lives: number
	config: Config
}

File: game/player.go
package game

import (
	"github.com/tjbearse/robo/game/cards"
	"github.com/tjbearse/robo/game/coords"
)

// TODO collapse player and Robot?
type Player struct {
	Name string
	Robot Robot
	Spawn SpawnSetting
	FlagNum int // i.e. the flag currently targetting
}

type Robot struct {
	Damage int
	Lives int
	Board []*cards.Card
	Configuration *coords.Configuration
}

type SpawnState int
const (
	Unset SpawnState = iota
	Rotatable
)
type SpawnSetting struct {
	State SpawnState
	Coord coords.Coord
}
